### PR TITLE
move to jenkins version with getJsonSignatureValidator() method

### PR DESCRIPTION
**Pull request by lanwen**
New versions of Jenkins don't work with current implementation of this plugin. They don't call those methods and don't writes certs

( jenkinsci/jenkins@86ea65a )

[Review on Reviewable](https://reviewable.io/reviews/jenkinsci/update-sites-manager-plugin/3)

### CODE REVIEW

File: src/test/java/jp/ikedam/jenkins/plugins/updatesitesmanager/internal/ExtendedCertJsonSignValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    public void shouldAddCustomCertToTrustAnchors() throws Exception {
+        String cert = IOUtils.toString(getClass().getClassLoader()
+                .getResourceAsStream("jp/ikedam/jenkins/plugins/updatesitesmanager/" +
+                        "ManagedUpdateSiteJenkinsTest/caCertificate.crt"), Charsets.UTF_8);
+        CertificateFactory cf = CertificateFactory.getInstance("X509");
+
+        Set<TrustAnchor> test = new ExtendedCertJsonSignValidator("test", cert).loadTrustAnchors(cf);
+        assertThat(test, hasSize(1));
+    }
+}
```
Comment (KostyaSha) [inline src/test/java/jp/ikedam/jenkins/plugins/updatesitesmanager/internal/ExtendedCertJsonSignValidatorTest.java]: NL ;)

**Comment by KostyaSha** [comment]
:+1:

**Comment by oleg-nenashev** [comment]
Definitely :+1: 
It also deserves a JIRA issue.

> New versions of Jenkins don't work with current implementation of this plugin. They don't call those methods and don't writes certs

It smells like a regression in API, doesn't it? CC @stephenc

**Comment by lanwen** [comment]
By default Jenkins now uses server-based download which directly gets update center without callback.
This PR works well with any kind of download - server/browser based

File: pom.xml [hunk 2]
```diff
@@ -3,7 +3,7 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>1.466</version><!-- which version of Jenkins is this plugin built against? -->
+    <version>1.609.3</version><!-- which version of Jenkins is this plugin built against? -->
```
Comment (ikedam) [inline pom.xml]: 1.596 is better, the baseline of the least LTS.

File: pom.xml [hunk 3]
```diff
@@ -55,16 +55,17 @@
...
+            <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>1.7</source>
+                    <target>1.7</target>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
```
Comment (ikedam) [inline pom.xml]: What is this for?

File: src/main/java/jp/ikedam/jenkins/plugins/updatesitesmanager/ManagedUpdateSite.java [hunk 4]
```diff
@@ -130,24 +127,6 @@ public String getNote()
...
-    protected File getCaCertificateFile()
-    {
-        File caCertificateDir = new File(Jenkins.getInstance().getRootDir(), "update-center-rootCAs");
-        if(!caCertificateDir.exists())
-        {
-            assert(caCertificateDir.mkdir());
-        }
-        
-        return new File(caCertificateDir, String.format("tmp-ManageUpdateSite-%s.crt", getId()));
-    }
```
Comment (ikedam) [inline src/main/java/jp/ikedam/jenkins/plugins/updatesitesmanager/ManagedUpdateSite.java]: It's ok though this breaks the binary compatibility.
I don't think other codes accessing this method.
(actually, I'm not sure why I had this method `protected`)

File: src/test/java/jp/ikedam/jenkins/plugins/updatesitesmanager/DescribedUpdateSiteJenkinsTest.java [hunk 5]
```diff
@@ -172,7 +172,6 @@ public void testDoConfigure() throws Exception
         // duplicate id
         {
             WebClient wc = new WebClient();
-            wc.setPrintContentOnFailingStatusCode(false);
```
Comment (ikedam) [inline src/test/java/jp/ikedam/jenkins/plugins/updatesitesmanager/DescribedUpdateSiteJenkinsTest.java]: I'm not sure why you remeved this.
I put this line to suppress unuseful outputs (they should be saved to xxxx-output.txt).
(I think I have to see how that behavior changes)

File: src/test/java/jp/ikedam/jenkins/plugins/updatesitesmanager/ManagedUpdateSiteJenkinsTest.java [hunk 6]
```diff
@@ -99,78 +94,6 @@ private File getResource(String name) throws URISyntaxException, FileNotFoundExc
         return new File(url.toURI());
     }
     
-    public void testDoPostBack() throws URISyntaxException, IOException, GeneralSecurityException
```
Comment (ikedam) [inline src/test/java/jp/ikedam/jenkins/plugins/updatesitesmanager/ManagedUpdateSiteJenkinsTest.java]: Though we no longer override `doPostBack`, this test is still valid as an integration.
Is it that this test does't work with this change?

File: pom.xml (hunk 3 above)
Comment (KostyaSha) [inline pom.xml]: Old core enforces 1.5 or 1.6 java level that has no sense today. (In my plugins i even jumped to 1.8 to get streams ;)

Comment (lanwen) [inline pom.xml]: for try-with-resources in ExtendedCertJsonSignValidator

File: pom.xml (hunk 2 above)
Comment (lanwen) [inline pom.xml]: I've get (current LTS - 1) version. Its a half of a year.

File: src/test/java/jp/ikedam/jenkins/plugins/updatesitesmanager/DescribedUpdateSiteJenkinsTest.java (hunk 5 above)
Comment (lanwen) [inline src/test/java/jp/ikedam/jenkins/plugins/updatesitesmanager/DescribedUpdateSiteJenkinsTest.java]: new versions of jenkins don't use custom fork of webclient and this method is removed

File: src/test/java/jp/ikedam/jenkins/plugins/updatesitesmanager/ManagedUpdateSiteJenkinsTest.java (hunk 6 above)
Comment (lanwen) [inline src/test/java/jp/ikedam/jenkins/plugins/updatesitesmanager/ManagedUpdateSiteJenkinsTest.java]: new jenkins versions removed 
```
WebRequestSettings wrs = new WebRequestSettings(
 -                    new URL(String.format("%s%s/byId/%s/postBack",
 -                            wc.getContextPath(),
 -                            Jenkins.getInstance().getUpdateCenter().getSearchUrl(),
 -                            target.getId()
 -                    )),
 -                    HttpMethod.POST
 -            );
```

so all the test don't work. I can't find an alternative to WebRequestSettings

**Comment by ikedam** [comment]
@lanwen 
I added some commits and created #5
(almost all changes are for test codes).
Would you review that?

**Pull request merged**